src/components/picker/types.tsx

```diff
@@ -95,6 +95,10 @@ export type PickerBaseProps = Omit<NewTextFieldProps, 'value' | 'onChange'> & {
     itemProps: PickerItemProps & {isSelected: boolean; isItemDisabled: boolean},
     label?: string
   ) => React.ReactElement;
+  /**
+   * A callback to be used when an item has been pressed, will stop selection if false is returned
+   */
+  onItemPress?: (item: PickerValue) => Promise<boolean>;
```

Comment:
assuming people dont need onItemPress to be asynchronous, typing wise they will still need to return a promise and that can be annoying, no?
if my logic is synchronous, TS won't complain?